frontend/src/pages/SharePage/SharePage.tsx

```diff
@@ -25,7 +26,7 @@ export const SharePage  = () => {
 
   const handleUnshare = async () => {
     await shares.delete({deviceID, email: userName})
-    history.push(`/devices/${deviceID}/users`)
+    history.push(location.pathname.replace(userName ? `/${userName}` : '/share', ''))
```

Comment:
This all looks good! Only thing is could we change the references to userName to email? We will soon be working on allowing users to have a username, but right now they only are able to have an email. I'm worried if you referer to emails as usernames that it could cause confusion in the future. Thanks!

```suggestion
    history.push(location.pathname.replace(email ? `/${email}` : '/share', ''))
```

Reply:
frontend/src/components/DeviceShareContainer/SharingForm.tsx was updated too because is using paramName